tests/scanner/audit/enabled_apis_rules_engine_test.py

```diff
@@ -0,0 +1,195 @@
+        self.assertEquals(0, len(list(violations)))
+
+        # Non-whitelisted APIs.
+        violations = list(rules_engine.find_policy_violations(
+            self.proj_3,
+            ['alpha.googleapis.com', 'bar.googleapis.com', 'other-api.com']))
+        self.assertEquals(1, len(violations))
+        self.assertEquals('ENABLED_APIS_ADDED_VIOLATION',
+                          violations[0].violation_type)
+        self.assertEquals(('alpha.googleapis.com', 'other-api.com'),
+                          violations[0].apis)
+
+        # API is whitelisted for Organization, but not globally (wildcard).
+        violations = list(rules_engine.find_policy_violations(
+            self.proj_1, ['qux.googleapis.com']))
+        self.assertEquals(1, len(violations))
+        self.assertEquals('ENABLED_APIS_ADDED_VIOLATION',
+                          violations[0].violation_type)
+        self.assertEquals(('qux.googleapis.com',), violations[0].apis)
+
+    def test_find_blacklist_violation(self):
+        """Test blacklist rules."""
+        rules_local_path = get_datafile_path(__file__,
+                                             'enabled_apis_test_rules_1.yaml')
+        rules_engine = eare.EnabledApisRulesEngine(
+            rules_file_path=rules_local_path)
+        rules_engine.build_rule_book()
+        self.assertEqual(4, len(rules_engine.rule_book.resource_rules_map))
+
+        # Everything is allowed.
+        violations = rules_engine.find_policy_violations(
+            self.proj_1, ['foo.googleapis.com', 'baz.googleapis.com'])
+        self.assertEquals(0, len(list(violations)))
+
+        # Blacklisted APIs.
+        violations = list(rules_engine.find_policy_violations(
+            self.proj_1,
+            ['foo.googleapis.com', 'bar.googleapis.com', 'baz.googleapis.com']))
+        self.assertEquals(1, len(violations))
+        self.assertEquals('ENABLED_APIS_ADDED_VIOLATION',
+                          violations[0].violation_type)
+        self.assertEquals(('bar.googleapis.com',), violations[0].apis)
+
+    def test_find_required_violation(self):
+        """Test required api rules."""
+        rules_local_path = get_datafile_path(__file__,
+                                             'enabled_apis_test_rules_2.yaml')
+        rules_engine = eare.EnabledApisRulesEngine(
+            rules_file_path=rules_local_path)
+        rules_engine.build_rule_book()
+        self.assertEqual(2, len(rules_engine.rule_book.resource_rules_map))
+
+        # Required API is included.
+        violations = rules_engine.find_policy_violations(
+            self.proj_3, ['foo.googleapis.com', 'bar.googleapis.com'])
+        self.assertEquals(0, len(list(violations)))
+
+        # Required API is missing.
+        violations = list(rules_engine.find_policy_violations(
+            self.proj_3, ['foo.googleapis.com']))
+        self.assertEquals(1, len(violations))
+        self.assertEquals('ENABLED_APIS_REMOVED_VIOLATION',
```

Comment:
If you still need to test for this type, you can make it a public constant in the app code, and then import it here for re-use.

Reply:
Fixed.